core/variant/variant.cpp

```diff
@@ -3239,8 +3239,11 @@ uint32_t Variant::recursive_hash(int recursion_count) const {
 	return 0;
 }
 
+#define hash_compare_scalar_base(p_lhs, p_rhs, semantic_comparison) \
+	(((p_lhs) == (p_rhs)) || (Math::is_nan(p_lhs) && Math::is_nan(p_rhs) && semantic_comparison))
```

Comment:
```suggestion
	(((p_lhs) == (p_rhs)) || (semantic_comparison && Math::is_nan(p_lhs) && Math::is_nan(p_rhs)))
```
Utilise short-circuit to avoid more expensive checks

Reply:
Good catch. Done.